array/array.go

```diff
@@ -0,0 +1,253 @@
+package array
+
+import (
+	"github.com/apache/arrow/go/arrow"
+	"github.com/apache/arrow/go/arrow/array"
+	"github.com/apache/arrow/go/arrow/memory"
+)
+
+//go:generate -command tmpl ../gotool.sh github.com/benbjohnson/tmpl
+//go:generate tmpl -data=@numeric.tmpldata -o numeric.gen.go numeric.gen.go.tmpl
+
+// Interface represents an immutable sequence of values.
+type Interface = array.Interface
+
+// Builder provides an interface to build arrow arrays.
+type Builder interface {
```

Comment:
Might be good to note here that you are mimicking Arrow's builder interface, albeit without the unexported methods.